### ISSUE TITLE
vulkan on android build instructions? 

### ISSUE THREAD

**Issue by stopthinking102** [ggml-org/whisper.cpp#2370]
has anyone tried building vulkan on android and done benchmarks on say adreno gpu 540.

**Comment by ryanheise** [comment]
No. FYI, I built on Arch Linux, and I set compile/target SDK to 36.

(Just to add for completeness, I was testing the turbo model for the benchmark.)

**Comment by AndreyPavlenko** [comment]
I was able to compile , but it does not work. It crashes here:
```sh
/vendor/lib/hw/vulkan.adreno.so (qglinternal::vkCmdBindPipeline(VkCommandBuffer_T*, VkPipelineBindPoint, VkPipeline_T*)+) ()
XXbase.apk!libwhisper_jni.so (offset XX) (ggml_backend_vk_graph_compute(ggml_backend*, ggml_cgraph*)+) ()
XXbase.apk!libwhisper_jni.so (offset XX) (ggml_backend_sched_graph_compute+) ()
XXbase.apk!libwhisper_jni.so (offset XX) (ggml_graph_compute_helper(ggml_backend_sched*, ggml_cgraph*, int, bool)+) ()
XXbase.apk!libwhisper_jni.so (offset XX) (whisper_decode_internal(whisper_context&, whisper_state&, whisper_batch const&, int, bool, bool (*)(void*), void*)+) ()
XXbase.apk!libwhisper_jni.so (offset XX) (whisper_full_with_state+) ()
XXbase.apk!libwhisper_jni.so (offset XX) (whisper_full+) ()
```

**Comment by AndreyPavlenko** [comment]
> FetchContent_Declare(vk_headers GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers GIT_TAG v1.3.268 GIT_SHALLOW TRUE)
> FetchContent_MakeAvailable(vk_headers)

You don't need to fetch the headers. You already have them with NDK. But you need the cpp bindings of the same version:
```bash
$ grep VK_HEADER_VERSION $ANDROID_NDK/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/vulkan/vulkan_core.h 
#define VK_HEADER_VERSION 275
#define VK_HEADER_VERSION_COMPLETE VK_MAKE_API_VERSION(0, 1, 3, VK_HEADER_VERSION)
```
The required version is 1.3.275.
```cmake
FetchContent_Declare(vk_hpp GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Hpp GIT_TAG v1.3.275 GIT_SHALLOW TRUE)
FetchContent_MakeAvailable(vk_hpp)
set(Vulkan_INCLUDE_DIR "${ANDROID_NDK}/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include" CACHE PATH "Vulkan Include" FORCE)
```

**Comment by guranu** [comment]
> I was able to compile , but it does not work on Adreno 830. It crashes here:
> 
> /vendor/lib/hw/vulkan.adreno.so (qglinternal::vkCmdBindPipeline(VkCommandBuffer_T*, VkPipelineBindPoint, VkPipeline_T*)+) ()
> XXbase.apk!libwhisper_jni.so (offset XX) (ggml_backend_vk_graph_compute(ggml_backend*, ggml_cgraph*)+) ()
> XXbase.apk!libwhisper_jni.so (offset XX) (ggml_backend_sched_graph_compute+) ()
> XXbase.apk!libwhisper_jni.so (offset XX) (ggml_graph_compute_helper(ggml_backend_sched*, ggml_cgraph*, int, bool)+) ()
> XXbase.apk!libwhisper_jni.so (offset XX) (whisper_decode_internal(whisper_context&, whisper_state&, whisper_batch const&, int, bool, bool (*)(void*), void*)+) ()
> XXbase.apk!libwhisper_jni.so (offset XX) (whisper_full_with_state+) ()
> XXbase.apk!libwhisper_jni.so (offset XX) (whisper_full+) ()

Same thing when compiling through termux.

**Comment by wjgong001** [comment]
I'm running whisper.cpp on a Motorola Android phone via Termux — no Vulkan, just CPU with NEON. On 8 cores with 12GB RAM:

• tiny model: 11s audio → 1.5s (real-time 7x)
• small model: ~10min for 15min audio

Compilation was straightforward: `git clone` → `make -j$(nproc)`. No need for Vulkan unless you're trying to run large models. If you just want transcription on-device, CPU-only works fine for tiny/small.

I documented the full process here including audio preprocessing and the comparison of Python vs C++ approaches (pip install openai-whisper fails on Termux, whisper.cpp works): https://github.com/wjgong001/termux-ai-toolkit/blob/main/docs/02-voice-to-text.md

For Vulkan specifically — you'll need the `clvk` or `vulkan-loader` packages from Termux, then compile whisper.cpp with `cmake -DWHISPER_VULKAN=ON`. But the performance gain on mobile GPU may not justify the complexity for most use cases.

**Comment by npodbielski** [comment]
Yes I know. It worked for me, but results for Polish were... ok... It was better for bigger versions, but slow. I wanted to try it with vulkan just of curiosity if this will be viable to do quicker with better results. For now it is not viable IMHO.
